src/adapters/class-jw-player.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * WP Video Sync: JW Player Adapter.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync\Adapters;
+
+use Alley\WP\WP_Video_Sync\API\JW_Player_API;
+use Alley\WP\WP_Video_Sync\Interfaces\Adapter;
+use DateTimeImmutable;
+use stdClass;
+
+/**
+ * JW Player Adapter.
+ */
+class JW_Player implements Adapter {
+
+	/**
+	 * The date of the last modification to the last batch of videos.
+	 *
+	 * @var ?DateTimeImmutable
+	 */
+	private ?DateTimeImmutable $last_modified_date;
```

Comment:
I can't figure out how this value would be read before it's initialized, as you indicated in your comment, based on my read of the code, but you could initialize it to `null` so it has a value which is compatible with the declared type to avoid this if it's still a problem.

Reply:
See my reply to the other comment